### ISSUE TITLE
Unsatisfied secure compilation options -fPIE -pie

### ISSUE THREAD

**Issue by JunXie-ZH** [vesoft-inc/nebula#5844]
I used a scanning tool to scan the nebula-graphd file and found Unsatisfied secure compilation options -fPIE -pie. I saw a related reply in the community: Nebula relies on static libraries and does not need this option. (https://discuss.nebula-graph.com.cn/t/topic/2846/2)

Is there a formal conclusion and reason for this?

**Comment by Shinji-IkariG** [comment]
In the current source code we use the -fPIC option, what scanning tool are you using? I want to reproduce this scan result.
Both nebula and nebula-third-party use the -fPIC option, so I want to know which part is showing up -fPIE

**Comment by JunXie-ZH** [comment]
Yes, my code exists -fPIE -pie .  But I used the security tool to scan nebula-graphd, the result showed that the -fPIE -pie not activated. I'd like to consult the reasons and the security risks.

`      -DCMAKE_C_FLAGS="-fstack-protector-all -D_FORTIFY_SOURCE=2 -O2 -Wl,-z,relro,-z,now,-z,noexecstack -s -fPIE -pie -fPIC -pie -Wl,--disable-new-dtags ${CMAKE_C_FLAGS}" \
      -DCMAKE_CXX_FLAGS="-fstack-protector-all -D_FORTIFY_SOURCE=2 -O2 -Wl,-z,relro,-z,now,-z,noexecstack -s -fPIE -pie -fPIC -pie -Wl,--disable-new-dtags ${CMAKE_CXX_FLAGS}"`

**Comment by JunXie-ZH** [comment]
Thanks! I would like to pick your brains with a few final questions.

1. Except for modifying the source code, is it impossible to make compiled binary file with `pie` because of the `-no-pie` linker option?

2. Which file in the repository does the `-no-pie` linker option exist in?
